src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -357,6 +357,66 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
             return expandedBytes;
         }
 
+        private static unsafe string UrlEncodeInternal(string originalString)
+        {
+            fixed (char* pch = originalString)
+            {
+                int safeCharCount = 0;
+                int spaceCharCount = 0;
+                for (int i = 0; i < originalString.Length; i++)
+                {
+                    char c = pch[i];
+                    if (c == ' ')
+                    {
+                        spaceCharCount++;
+                    }
+                    else if (IsUrlSafeChar(c))
+                    {
+                        safeCharCount++;
+                    }
+                }
+
+                if (spaceCharCount == 0 && safeCharCount == originalString.Length)
+                {
+                    // Nothing to expand
+                    return originalString;
+                }
+                
+                int unsafeByteCount = Encoding.UTF8.GetByteCount(pch, originalString.Length) - safeCharCount - spaceCharCount;
```

Comment:
Understood,
I'm just wondering whether this is more or less performant (calculating the byte count for each char)
If you look above, @jkotas said that `Encoding.Utf8.GetByteCount` actually parses more than 1 char at a time, causing the performance to not suffer as much when strings get longer.
By calculating GetByteCount() for each unsafe char, are we forgoing that perf enhancement, and perhapss making things worse?
@jkotas do you have an opinion?